juseppe-cli/src/main/java/ru/lanwen/jenkins/juseppe/files/WatchFiles.java

```diff
@@ -61,12 +91,53 @@ public void run() {
             WatchKey key = watcher.take();
             while (key != null) {
 
-                if (key.pollEvents().stream().anyMatch(hasExt(".hpi").or(hasExt(".jpi")))) {
-                    LOG.trace("HPI (JPI) list modify found!");
-                    UpdateSiteGen.updateSite(props).withDefaults().toSave().saveAll();
+                Path dir = keys.get(key);
+
+                if (dir == null) {
+                    LOG.error(String.format("%s: WatchKey: %s is not recognized!", getClass(), key.toString()));
```

Comment:
one more thing - sorry for not noticed it earlier
slf4j supports placeholders with `{}` instead of `%s` - so you can omit `String.format`